source/core/slang-dictionary.h

```diff
@@ -167,7 +167,7 @@ namespace Slang
 					return FindPositionResult(hashPos, -1);
 				}
 				numProbes++;
-				hashPos = (hashPos + GetProbeOffset(numProbes)) & bucketSizeMinusOne;
```

Comment:
Shouldn't that be `bucketSize` or `(bucketSizeMinusOne + 1)` ? That anding if bucketSize is a power of 2, gives values between 0 and bucketSize - 1. Mod will give values (assuming bucketSize >=2) between 0 and  bucketSize - 1 - 1

Reply:
It seems that the bucketSize is a power of 2 still? So why not stick with & - it would be faster. Not using a power2 might provide better hash map storage distribution.